### PR TITLE
liquidprompt: remove sample config

### PR DESCRIPTION
**Pull request by augmentedfourth**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Liquidprompt is no longer distributing a `liquidpromptrc-dist` default configuration file, as it hasn't been maintained as well as documentation. The project has decided to direct people to directly reference the documentation (which I've done here). This change is currently breaking `--HEAD` installs due to the missing file. They're also developing a tool to generate a default config out of the documentation itself, the use of which I will add to the caveats once it's in an official release.

### CODE REVIEW

**Comment by chenrui333** [comment]
> This change is currently breaking `--HEAD` installs due to the missing file.

why not just remove HEAD build then, we can always bring it back later.

**Comment by augmentedfourth** [comment]
> > This change is currently breaking `--HEAD` installs due to the missing file.
> 
> why not just remove HEAD build then, we can always bring it back later.

Sorry; I guess I wasn't clear. The HEAD build is currently broken, but future full versions will break as well, because the file is not coming back. This is preemptive to adapt to a design change in the upstream project.

**Comment by augmentedfourth** [comment]
And I meant the *upstream* change is breaking installs; this Homebrew update is fixing that breakage. Both releases and `--HEAD` installs will work after this PR is implemented (I've tested both).

**Comment by chenrui333** [comment]
> And I meant the _upstream_ change is breaking installs; this Homebrew update is fixing that breakage. Both releases and `--HEAD` installs will work after this PR is implemented (I've tested both).

I see, that makes sense.

File: Formula/liquidprompt.rb [hunk 1]
```diff
@@ -10,21 +10,27 @@ class Liquidprompt < Formula
...
   def install
-    share.install "liquidpromptrc-dist"
     share.install "liquidprompt"
   end
 
-  def caveats
+  define_method :caveats do
+    if build.head?
+      doc_url = "https://liquidprompt.readthedocs.io/en/latest/config.html"
+      head_note = " (for the HEAD version)"
```
Comment (chenrui333) [inline Formula/liquidprompt.rb:22]: can you reference `#{Utils.git_head}` in here?

**Comment by chenrui333** [comment]
can you also amend your commit message as `liquidprompt: update docs`

File: Formula/liquidprompt.rb (hunk 1 above)
Comment (augmentedfourth) [reply Formula/liquidprompt.rb:22]: I'm not sure what you mean here. I used `build.head?` because I saw that in the official documentation.

**Comment by augmentedfourth** [comment]
> can you also amend your commit message as `liquidprompt: update docs reference`

Will do once I figure out what you mean in the other comment.

File: Formula/liquidprompt.rb [hunk 2]
```diff
@@ -10,21 +10,27 @@ class Liquidprompt < Formula
...
       Add the following lines to your bash or zsh config (e.g. ~/.bash_profile):
         if [ -f #{HOMEBREW_PREFIX}/share/liquidprompt ]; then
           . #{HOMEBREW_PREFIX}/share/liquidprompt
         fi
 
       If you'd like to reconfigure options, you may do so in ~/.liquidpromptrc.
-      A sample file you may copy and modify has been installed to
-        #{HOMEBREW_PREFIX}/share/liquidpromptrc-dist
+      Configuration documentation#{head_note} can be found here:
+        #{doc_url}
```
Comment (SMillerDev) [inline Formula/liquidprompt.rb:33]: I think we should just remove this if we'll just refer to upstream docs. Caveats are only for Homebrew documentation

Comment (augmentedfourth) [reply Formula/liquidprompt.rb:33]: It's not obvious how to set up or configure the tool, which is why we'd like to put that in front of the user at install time. In the future we may again be able to generate and provide a sample file based on the current state of the documentation.

Comment (SMillerDev) [reply Formula/liquidprompt.rb:33]: So, most people do not read the caveats and even less people will if we start using it for all kinds of documentation.

**Review by SMillerDev — APPROVED** [review]
Thanks @augmentedfourth   ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request merged**